packages/runtime/src/widgets/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,167 @@
+    const namedObj: Record<string, unknown> = {};
+    namedObj[itemTemplate.name] = value;
+    return namedObj;
+  });
+  const stepsLength = namedData.length;
+  const stepTypes = itemTemplate.template;
+  const handleNext = useCallback(() => {
+    if (activeStep < namedData.length - 1) {
+      setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    }
+  }, [activeStep, namedData.length]);
+
+  const handleBack = useCallback(() => {
+    if (activeStep !== 0) {
+      setActiveStep((prevActiveStep) => prevActiveStep - 1);
+    }
+  }, [activeStep]);
+  const handleStep = (step: number) => () => {
+    setActiveStep(step);
+  };
+  const { values } = useRegisterBindings({ ...props, activeStep }, props.id, {
+    handleNext,
+    handleBack,
+  });
+  return (
+    <div>
+      <MUIStepper
+        activeStep={values?.activeStep}
+        alternativeLabel
+        connector={
+          <CustomConnector
+            connector_color={props?.styles.connectorColor}
+            connector_height={props?.styles.connectorHeight}
+          />
+        }
+      >
+        {values?.["item-template"]
+          ? namedData?.map((data, index) => (
+              <Step key={index}>
+                <StepButton onClick={handleStep(index)}>
+                  <StepLabel
+                    StepIconComponent={(iconProps: StepOwnProps) => {
+                      const newProps = {
+                        stepTypes,
+                        ...(iconProps as StepIconProps),
+                        data,
+                        index: values.activeStep,
+                        stepsLength,
+                      };
+                      return CustomStepIcon(newProps);
+                    }}
+                  />
+                </StepButton>
+              </Step>
+            ))
+          : null}
+      </MUIStepper>
+      <div>
+        {values?.steps.map((step, index) => (
+          <div key={step.stepLabel}>
+            {index === values?.activeStep && (
+              <>
+                {step?.contentWidget
+                  ? EnsembleRuntime.render([step.contentWidget])
+                  : null}
+              </>
+            )}
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+};
+
+WidgetRegistry.register("Stepper", Stepper);
+
+const CustomConnector = styled(StepConnector)(
+  (props: CustomConnectorProps) => ({
+    [`&.${stepConnectorClasses.alternativeLabel}`]: {
+      top: 20,
+    },
+    [`&.${stepConnectorClasses.active}`]: {
+      [`& .${stepConnectorClasses.line}`]: {
+        borderColor: props?.connector_color ?? "grey",
+      },
+    },
+    [`&.${stepConnectorClasses.completed}`]: {
+      [`& .${stepConnectorClasses.line}`]: {
+        borderColor: "#31C48D",
+      },
+    },
+    [`& .${stepConnectorClasses.line}`]: {
+      borderColor: "#eaeaf0",
+      borderTopWidth: props?.connector_height ?? 3,
+      borderRadius: 1,
+    },
+  })
+);
+
+const CustomStepIcon = (
```

Comment:
What is the difference between this and `StepType` component?

Reply:
This can be called as a wrapper only where I am destructuring some data,  I think I can simplify it further